confluent_kafka/schema_registry/config.py

```diff
@@ -0,0 +1,295 @@
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Python 2 considers int an instance of str
+import json
+
+from .error import SchemaRegistryClientError
+
+try:
+    string_type = basestring  # noqa
+except NameError:
+    string_type = str
+
+from requests import utils, Session
+
+from ..config import Config
+from .schema_registry_client import SchemaRegistryClient
+from .serdes import SubjectNameStrategy, TopicRecordNameStrategy, TopicNameStrategy, \
+    RecordNameStrategy
+
+"""
+REST Client
+"""
+URL = 'url'
+REST_CLIENT = 'rest.client'
+"""
+Serializer Config
+"""
+SUBJECT_NAME_STRATEGY = 'subject.name.strategy'
+AUTO_REGISTRATION = 'auto.register.schemas'
+
+"""
+TLS configuration properties
+"""
+CA_LOCATION = 'ssl.ca.location'
+CERTIFICATE_LOCATION = 'ssl.certificate.location'
+KEY_LOCATION = 'ssl.key.location'
+
+"""
+Basic Auth configuration properties
+"""
+BASIC_AUTH_CREDENTIALS_SOURCE = 'basic.auth.credentials.source'
+BASIC_AUTH_USER_INFO = 'basic.auth.user.info'
+SASL_USERNAME = 'sasl.username'
+SASL_PASSWORD = 'sasl.password'
+SASL_MECHANISM = 'sasl.mechanism'
+
+
+class DefaultRestClient(Session):
+    default_headers = {'Accept': "application/vnd.schemaregistry.v1+json,"
+                                 " application/vnd.schemaregistry+json,"
+                                 " application/json"}
+
+    def __init__(self, conf):
+        super(DefaultRestClient, self).__init__()
+
+        self.base_url = conf.Url.rstrip('/')
+
+        if conf.Certificate is not (None, None):
+            self.cert = conf.Certificate
+
+        if conf.CA is not None:
+            self.verify = conf.CA
+
+        if conf.Credentials:
+            self.auth = conf.Credentials
+
+    def __del__(self):
+        self.close()
+
+    def GET(self, url, decoder=None):
+        return self.send_request(url, method='GET', decoder=decoder)
+
+    def POST(self, url, body, encoder=None, decoder=None, **kwargs):
+        return self.send_request(url, method='POST',
+                                 body=body, encoder=encoder, decoder=decoder,
+                                 **kwargs)
+
+    def DELETE(self, url):
+        return self.send_request(url, method='DELETE')
+
+    def PUT(self, url, body=None, encoder=None, decoder=None):
+        return self.send_request(url, method='PUT', body=body,
+                                 encoder=encoder, decoder=decoder)
+
+    def url_formatter(self, template, *args):
```

Comment:
This does not belong here it unnecessarily ties the SR API back to the rest client which I was trying to get away from. It's not that big of a deal to format the URL on the SR side and prepend the base here instead of trying to get it done in one go. Will fix